pkg/apis/monitoring/v1/types.go

```diff
@@ -150,6 +150,66 @@ func (pc *ProxyConfig) Validate() error {
 	return nil
 }
 
+type HTTPHeader struct {
+	// Name of the HTTP header.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name"`
+
+	// References to secret keys holding the header values.
+	//
+	// +kubebuilder:validation:MinItems=1
+	// +listType=set
+	// +optional
+	SecretRefs []v1.SecretKeySelector `json:"secretRefs,omitempty"`
```

Comment:
I would recommend that we set the field required too since a header name without a value is a no-operation.
```suggestion
	// +kubebuilder:validation:Required
	SecretRefs []v1.SecretKeySelector `json:"secretRefs"`
```